org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/persistence/git/GitProvider.xtend

```diff
@@ -47,7 +67,48 @@ class GitProvider {
 	}
 
 	private def Git initializeNew(File workspace) {
-		return Git.cloneRepository.setURI(config.remoteRepoUrl).setDirectory(workspace).call
+		val command = Git.cloneRepository => [
+			setURI(config.remoteRepoUrl)
+			setSshSessionFactory
+			setDirectory(workspace)
+		]
+		return command.call
 	}
 
+	private def <T, C extends GitCommand<T>> void setSshSessionFactory(TransportCommand<C, ?> command) {
+		
+		val sshSessionFactory = new JschConfigSessionFactory {
```

Comment:
Can this be made into a "proper" class in its own file, instead of implementing it in-place here as an anonymous inner class?